### ISSUE TITLE
Non-witness transactions cannot include issuances

### ISSUE THREAD

**Issue by instagibbs** [ElementsProject/elements#473]
Transaction serialization requires that witness-empty transactions have empty witness vectors for inputs and outputs. While fixing the OOB access in VerifyAmounts we accidentally softforked out the ability to have issuances in transactions with no other witness data. In real-life usage this is fairly unlikely, since any blinding, any segwit signatures, and any peg-in data make this not true.

related: https://github.com/ElementsProject/elements/pull/472

**Comment by instagibbs** [comment]
Hmm do we want to target fixing this for dynafed? We'd have to expose chain-state knowledge to `VerifyAmounts`, but otherwise would be simple. @stevenroose @apoelstra

**Comment by stevenroose** [comment]
You mean allowing this once the dynafed BIP9 deployment is activated?

**Comment by instagibbs** [comment]
Couple this change in with dynafed activation, yes. We could also just wait for a follow-up mega-HF which we will have to do at some point anyways.

**Comment by stevenroose** [comment]
Yeah since we have a HF wishlist anyways, I'd vote to keep the dynafed HF dynafed-only. With dynafed we'll have everything in place to do a more smooth actual consensus HF.
